tests/advanced_visibility_test.go

```diff
@@ -1949,3 +1956,407 @@ func (s *advancedVisibilitySuite) updateMaxResultWindow() {
 	}
 	s.FailNow(fmt.Sprintf("ES max result window size hasn't reach target size within %v", (numOfRetry*waitTimeInMs)*time.Millisecond))
 }
+
+// TODO(bergundy): Clean some of the duplication here with the commented tests below this once we have the matching functionality in place
+func (s *advancedVisibilitySuite) TestWorkerTaskReachability_WithoutTasks_ByBuildId() {
+	ctx := NewContext()
+	tq1 := s.T().Name()
+	tq2 := s.T().Name() + "-2"
+	tq3 := s.T().Name() + "-3"
+	v0 := s.T().Name() + "v0"
+	v01 := s.T().Name() + "v0.1"
+	var err error
+
+	_, err = s.engine.UpdateWorkerBuildIdCompatibility(ctx, &workflowservice.UpdateWorkerBuildIdCompatibilityRequest{
+		Namespace: s.namespace,
+		TaskQueue: tq1,
+		Operation: &workflowservice.UpdateWorkerBuildIdCompatibilityRequest_AddNewBuildIdInNewDefaultSet{
+			AddNewBuildIdInNewDefaultSet: v0,
+		},
+	})
+	s.Require().NoError(err)
+	_, err = s.engine.UpdateWorkerBuildIdCompatibility(ctx, &workflowservice.UpdateWorkerBuildIdCompatibilityRequest{
+		Namespace: s.namespace,
+		TaskQueue: tq2,
+		Operation: &workflowservice.UpdateWorkerBuildIdCompatibilityRequest_AddNewBuildIdInNewDefaultSet{
+			AddNewBuildIdInNewDefaultSet: v0,
+		},
+	})
+	s.Require().NoError(err)
+	_, err = s.engine.UpdateWorkerBuildIdCompatibility(ctx, &workflowservice.UpdateWorkerBuildIdCompatibilityRequest{
+		Namespace: s.namespace,
+		TaskQueue: tq1,
+		Operation: &workflowservice.UpdateWorkerBuildIdCompatibilityRequest_AddNewCompatibleBuildId{
+			AddNewCompatibleBuildId: &workflowservice.UpdateWorkerBuildIdCompatibilityRequest_AddNewCompatibleVersion{
+				ExistingCompatibleBuildId: v0,
+				NewBuildId:                v01,
+			},
+		},
+	})
+	s.Require().NoError(err)
```

Comment:
nit: This test reads slightly more obviously if this update happens after the first one so it's both tq1 ops, then tq2, then tq3

Reply:
SGTM.